examples/define-syntax-rule.kl

```diff
@@ -1,44 +1,40 @@
 #lang "prelude.kl"
 
+(import (rename (shift "prelude.kl" 1) [syntax-case raw-syntax-case]))
 (import (shift "prelude.kl" 1))
 (import (shift "do.kl" 1))
 (import (shift "list-syntax.kl" 1))
-(import (shift "quasiquote.kl" 1))
+(import (shift "dot-dot-dot.kl" 1))
 (import (shift "syntax.kl" 1))
 
 (define-macros
-  ((define-macro
+  ([define-macro
      (lambda (stx)
-       (syntax-case stx
-         ((list (_ pattern body))
-          (syntax-case pattern
-           ((cons macro-name args)
-            (pure `(define-macros
-                     ((,macro-name (lambda (stx)
-                                     (syntax-case stx
-                                      ((list ,pattern)
-                                       ,body)))))))))))))))
+       (syntax-case stx ()
+         [(_ (,macro-name ,args ...) ,body)
+          (pure `(define-macros
+                   ([,macro-name
+                     (lambda (stx)
+                       (raw-syntax-case stx
+                         [(list (_ ,args ...))
+                          ,body]))])))]))]))
 
 (define-macros
-  ((define-syntax-rule
+  ([define-syntax-rule
      (lambda (stx)
-       (syntax-case stx
-         ((list (_ pattern template))
-          (syntax-case pattern
-           ((cons macro-name args)
-            (do (unquoted-template <- (foldlM (lambda (t arg)
-                                                (replace-identifier
-                                                  arg
-                                                  (list-syntax ('unquote arg) stx)
-                                                  t))
-                                              template
-                                              args))
-                (quasiquoted-template <- (pure (list-syntax ('quasiquote unquoted-template) stx)))
-                (pure `(define-macros
-                         ((,macro-name (lambda (stx)
-                                         (syntax-case stx
-                                          ((list ,pattern)
-                                           (pure ,quasiquoted-template)))))))))))))))))
+       (syntax-case stx ()
+         [(_ (,macro-name ,args ...) ,template)
+          (do (unquoted-template <- (foldlM (lambda (t arg)
+                                              (replace-identifier arg `(,'unquote ,arg) t))
+                                            template
+                                            args))
+              (quasiquoted-template <- (pure `(,'quasiquote ,unquoted-template)))
```

Comment:
(pure ``,unquoted-template)

would have worked too, but only because I don't implement the [R6RS](http://www.r6rs.org/final/html/r6rs/r6rs-Z-H-14.html#node_sec_11.17) spec for nested quotations, according to which the result of

    (let ([x 'foo])
      ``,x)

should be

    `,x

not

    `foo